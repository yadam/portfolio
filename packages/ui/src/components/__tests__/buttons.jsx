import React from 'react';
import renderer from 'react-test-renderer';
import Buttons from '../buttons';

let mockProps;

describe('Buttons', () => {
  beforeEach(() => {
    mockProps = {
      onChange: jest.fn(),
      reaction: undefined,
    };
  });

  it('renders correctly when no reaction has been recorded', () => {
    const tree = renderer.create(<Buttons {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when a funny reaction has been recorded', () => {
    mockProps.reaction = 'funny';
    const tree = renderer.create(<Buttons {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when a meh reaction has been recorded', () => {
    mockProps.reaction = 'meh';
    const tree = renderer.create(<Buttons {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
