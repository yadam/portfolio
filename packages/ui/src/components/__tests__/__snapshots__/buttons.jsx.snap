// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders correctly when a funny reaction has been recorded 1`] = `
<div
  className="MuiToggleButtonGroup-root-4 MuiToggleButtonGroup-selected-5 Buttons-buttonGroup-2"
>
  <button
    className="MuiButtonBase-root-10 MuiToggleButton-root-6 MuiToggleButton-selected-8 Buttons-button-1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
    value="funny"
  >
    <span
      className="MuiToggleButton-label-9"
    >
      <h3
        className="MuiTypography-root-13 MuiTypography-h3-27 Buttons-faces-3"
      >
        <span
          aria-label="That's Funny!"
          role="img"
        >
          ðŸ˜†
        </span>
      </h3>
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </button>
  <button
    className="MuiButtonBase-root-10 MuiToggleButton-root-6 Buttons-button-1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
    value="meh"
  >
    <span
      className="MuiToggleButton-label-9"
    >
      <h3
        className="MuiTypography-root-13 MuiTypography-h3-27 Buttons-faces-3"
      >
        <span
          aria-label="Meh..."
          role="img"
        >
          ðŸ˜’
        </span>
      </h3>
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </button>
</div>
`;

exports[`Buttons renders correctly when a meh reaction has been recorded 1`] = `
<div
  className="MuiToggleButtonGroup-root-4 MuiToggleButtonGroup-selected-5 Buttons-buttonGroup-2"
>
  <button
    className="MuiButtonBase-root-10 MuiToggleButton-root-6 Buttons-button-1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
    value="funny"
  >
    <span
      className="MuiToggleButton-label-9"
    >
      <h3
        className="MuiTypography-root-13 MuiTypography-h3-27 Buttons-faces-3"
      >
        <span
          aria-label="That's Funny!"
          role="img"
        >
          ðŸ˜†
        </span>
      </h3>
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </button>
  <button
    className="MuiButtonBase-root-10 MuiToggleButton-root-6 MuiToggleButton-selected-8 Buttons-button-1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
    value="meh"
  >
    <span
      className="MuiToggleButton-label-9"
    >
      <h3
        className="MuiTypography-root-13 MuiTypography-h3-27 Buttons-faces-3"
      >
        <span
          aria-label="Meh..."
          role="img"
        >
          ðŸ˜’
        </span>
      </h3>
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </button>
</div>
`;

exports[`Buttons renders correctly when no reaction has been recorded 1`] = `
<div
  className="MuiToggleButtonGroup-root-4 MuiToggleButtonGroup-selected-5 Buttons-buttonGroup-2"
>
  <button
    className="MuiButtonBase-root-10 MuiToggleButton-root-6 Buttons-button-1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
    value="funny"
  >
    <span
      className="MuiToggleButton-label-9"
    >
      <h3
        className="MuiTypography-root-13 MuiTypography-h3-27 Buttons-faces-3"
      >
        <span
          aria-label="That's Funny!"
          role="img"
        >
          ðŸ˜†
        </span>
      </h3>
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </button>
  <button
    className="MuiButtonBase-root-10 MuiToggleButton-root-6 Buttons-button-1"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
    value="meh"
  >
    <span
      className="MuiToggleButton-label-9"
    >
      <h3
        className="MuiTypography-root-13 MuiTypography-h3-27 Buttons-faces-3"
      >
        <span
          aria-label="Meh..."
          role="img"
        >
          ðŸ˜’
        </span>
      </h3>
    </span>
    <span
      className="MuiTouchRipple-root-49"
    />
  </button>
</div>
`;
